### ISSUE TITLE
Improve UI for quicker access to branches

### ISSUE THREAD

**Issue by Himura2la** [gitahead/gitahead#513]
My typical git workflow is the following:

1. Pull the `dev` branch
2. **Create a new branch**
3. Commit and Push
4. **Probably switch to another branch to test someone else's changes and go back**
5. Add some more commits
6. **Get back to a `dev` branch** (would be great to have it on `origin/dev` before)

And I have problems with the **highlighted** items (2,4,6). I spend too much time digging through the tree in search of the right branch, and I don't like to use context menu or menubar at all, it requires too many highly-aimed clicks.

It would be great to have a branch list and a "new branch" button in a main window.

For branch list, I suggest using this empty area:

![image](https://user-images.githubusercontent.com/7802808/95716321-f704ee80-0c73-11eb-9e0a-82ae4d82343d.png)

The "new branch" button could be on a toolbar, or as a plus button in the branch list/

**Comment by Himura2la** [comment]
OK, I found that branch list is accessible above the tree. Checking one itam

![image](https://user-images.githubusercontent.com/7802808/95741781-8f609a80-0c97-11eb-80ff-f9293d294284.png)

**Comment by viking1304** [comment]
That "empty area" is maybe empty for you, but not for us with 50+ repositories. That is place for repositories, not for branches.

Branches are above commits, but you need to expand that area first. When you click on that arrow below gear icon, you will see the list of local branches.

![image](https://user-images.githubusercontent.com/28989870/99991264-1afc3980-2db5-11eb-87bc-0b549861ff09.png)

To add new local branch, select New Branch from Branch menu

![image](https://user-images.githubusercontent.com/28989870/99991772-bd1c2180-2db5-11eb-9d43-25e92c410eea.png)

Choose start point branch and click on Create

![image](https://user-images.githubusercontent.com/28989870/99993042-6dd6f080-2db7-11eb-98cb-9d7b7ae31b11.png)

And you just created new local branch that is same as develop

![image](https://user-images.githubusercontent.com/28989870/99991983-02405380-2db6-11eb-8599-5d7804dfeaf3.png)

To create new local branch from remote, select Remotes tab, than desired branch and finally click New Local Branch from right click menu

![image](https://user-images.githubusercontent.com/28989870/99992834-294b5500-2db7-11eb-874f-91585036b750.png)
 
When you double click any local branch, you will be switched to that branch. It should do just that - switch active branch. Nothing more, nothing less.

So, for point 2. you have to select appropriate option from navigation, instead of a button click. It's not one click, but should still be acceptable for everyone. On the other hand, I would not mind to have New Local Branch in right click menu of any local branch, that would indeed be faster.

For point 4 and 6 - just double click desired branch in Local Branches list.

I hope this helps, so please leve your "empty area" alone, because some of us need even more space there. :)

**Comment by Himura2la** [comment]
OK, I'm fine with the branch list, having it always expanded works for me. 

As for new branches, I know how to create a new branch and I think it's not convenient. I believe there should be a separate toolbar button for new branch. Why stashing has a button group, but branching has not? Stashes are required much less often than branches.

For point 4 and 6 - I don't want to search for the corresponding item in another list if I already found it in the tree. Double-clicking in the tree is not implemented at all, why? I see no point in this...

**Comment by Himura2la** [comment]
I changed the issue, bump

**Comment by wiegell** [comment]
I agree that this should be addressed. I would like to add that when using the checkout menu (or cmd + shit + h) it should be possible to start writing right away without having to do a mouseclick.
Switching between branches is an essential operation and clicks should be minimised.
